benchmarks/benchmarks/solarposition.py

```diff
@@ -33,11 +36,51 @@ def time_ephemeris_localized(self):
         solarposition.ephemeris(self.times_localized, self.lat, self.lon)
 
     def time_spa_python(self):
-        solarposition.spa_python(self.times_localized[::5], self.lat, self.lon)
+        solarposition.spa_python(self.times_localized, self.lat, self.lon)
+
+    def time_pyephem(self):
+        solarposition.pyephem(self.times_localized, self.lat, self.lon)
 
     def time_sun_rise_set_transit_spa(self):
-        sun_rise_set_transit_spa(self.times_localized[::30],
-                                 self.lat, self.lon)
+        sun_rise_set_transit_spa(self.times_daily, self.lat, self.lon)
+
+    def time_sun_rise_set_transit_ephem(self):
+        solarposition.sun_rise_set_transit_ephem(
+            self.times_daily, self.lat, self.lon)
+
+    def time_sun_rise_set_transit_geometric(self):
+        dayofyear = self.times_daily.dayofyear
+        declination = solarposition.declination_spencer71(dayofyear)
+        equation_of_time = solarposition.equation_of_time_spencer71(dayofyear)
+        solarposition.sun_rise_set_transit_geometric(
+            self.times_daily, self.lat, self.lon, declination,
+            equation_of_time)
```

Comment:
I'm not sure what to think about running more than one function in a single benchmark.  It would be a little awkward/repetitive to run benchmark each step of a sequence individually, but it seems like the rule of thumb "only test one thing at a time" for unit tests ought to apply to benchmarks as well.  

That said, any benchmark is better than no benchmark!

Reply:
Also consider calculating `dayofyear` in the setup function because it's external

Reply:
In general I agree. In this case I wanted a more direct comparison to the other `sun_rise_set_transit` functions, so I put all that in the test. The `dayofyear` calculation is part of a fair comparison. But I also see the argument to test each of them individually and add them up. For now, I renamed the existing function to make it clear it's for comparison.